packages/framework/src/Form/Constraints/DeliveryAddressOfCurrentCustomerValidator.php

```diff
@@ -34,16 +34,18 @@ public function validate($deliveryAddress, Constraint $constraint): void
             throw new \Symfony\Component\Validator\Exception\UnexpectedTypeException($constraint, UniqueCollection::class);
         }
 
-        if ($deliveryAddress !== null) {
-            $customerUser = $this->currentCustomerUser->findCurrentCustomerUser();
-
-            if ($customerUser !== null) {
-                if (!in_array($deliveryAddress, $customerUser->getCustomer()->getDeliveryAddresses(), true)) {
-                    $this->context->addViolation($constraint->message);
-                }
-            } else {
+        if ($deliveryAddress === null) {
+            return;
+        }
+
+        $customerUser = $this->currentCustomerUser->findCurrentCustomerUser();
+
+        if ($customerUser !== null) {
+            if (!in_array($deliveryAddress, $customerUser->getCustomer()->getDeliveryAddresses(), true)) {
                 $this->context->addViolation($constraint->message);
             }
+        } else {
+            $this->context->addViolation($constraint->message);
```

Comment:
Maybe this will do even better

```php
if (
    $customerUser !== null
    && in_array($deliveryAddress, $customerUser->getCustomer()->getDeliveryAddresses(), true)
) {
    return;
}

$this->context->addViolation($constraint->message);
```

which may be transformed to 

```php
if (
    $customerUser === null
    || !in_array($deliveryAddress, $customerUser->getCustomer()->getDeliveryAddresses(), true)
) {
    $this->context->addViolation($constraint->message);
}
```

But still I am not happy about checking if $customer is found... It should be handled via exception as we discussed recently.

Reply:
OT: In this particular case, I think the last code you wrote is better than the exception ;) 
```php
try {
    $customerUser = $this->currentCustomerUser->getCurrentCustomerUser();
    if (!in_array($deliveryAddress, $customerUser->getCustomer()->getDeliveryAddresses(), true)) {
       $this->context->addViolation($constraint->message);
    }
} catch (UserNotFoundException $ex) {
    $this->context->addViolation($constraint->message);
}
```

I leave up to you whether you want to apply changes you suggested as the PR is not about refactoring.

Reply:
Does not make sense to leave this unchanged when the discussion is already in progress.

Changed in 81bd90ec105b347d7055b4e2a2b477a16a329838

Reply:
cool